server/parsec/components/memory/datamodel.py

```diff
@@ -227,6 +244,181 @@ def active_sequester_services(self) -> Iterable[MemorySequesterService]:
+                )
+
+        for device in self.devices.values():
+            if redacted:
+                common_certificates_unordered.append(
+                    (device.cooked.timestamp, 1, device.redacted_device_certificate, device.cooked)
+                )
+            else:
+                common_certificates_unordered.append(
+                    (device.cooked.timestamp, 1, device.device_certificate, device.cooked)
+                )
+
+        for ts, _, raw, cooked in sorted(common_certificates_unordered, key=lambda x: (x[0], x[1])):
+            yield (ts, raw, cooked)
+
+    def ordered_sequester_certificates(
+        self,
+    ) -> Iterator[tuple[DateTime, bytes, SequesterTopicCertificate]]:
+        """
+        Return all sequester certificates ordered by timestamp.
+        """
+        if self.sequester_authority_certificate is None:
+            return
+
+        assert self.cooked_sequester_authority is not None
+        assert self.sequester_services is not None
+
+        yield (
+            self.cooked_sequester_authority.timestamp,
+            self.sequester_authority_certificate,
+            self.cooked_sequester_authority,
+        )
+
+        sequester_services_unordered: list[tuple[DateTime, bytes, SequesterTopicCertificate]] = []
+        for service in self.sequester_services.values():
+            sequester_services_unordered.append(
+                (service.cooked.timestamp, service.sequester_service_certificate, service.cooked)
+            )
+            if service.cooked_revoked:
+                assert service.sequester_revoked_service_certificate is not None
+                sequester_services_unordered.append(
+                    (
+                        service.cooked_revoked.timestamp,
+                        service.sequester_revoked_service_certificate,
+                        service.cooked_revoked,
+                    )
+                )
+
+        yield from sorted(sequester_services_unordered, key=lambda x: x[0])
+
+    def ordered_realm_certificates(
+        self, realm_id: VlobID
+    ) -> Iterator[tuple[DateTime, bytes, RealmTopicCertificate]]:
+        """
+        Return all realm certificates ordered by timestamp.
+        """
+        realm = self.realms[realm_id]
+
+        # Collect all the certificates related to the realm
+        realm_certificates_unordered = []
+        realm_certificates_unordered += [
+            (role.cooked.timestamp, role.realm_role_certificate, role.cooked)
+            for role in realm.roles
+        ]
+        realm_certificates_unordered += [
+            (role.cooked.timestamp, role.realm_key_rotation_certificate, role.cooked)
+            for role in realm.key_rotations
+        ]
+        realm_certificates_unordered += [
+            (role.cooked.timestamp, role.realm_name_certificate, role.cooked)
+            for role in realm.renames
+        ]
+        # TODO: support archiving here !
```

Comment:
If this comment references the workspace archiving and deletion feature, I would add the issue number.
```suggestion
        # TODO #6092: support archiving here !
```

Reply:
:+1:  (I've also updated the other TODOs mentioning archiving support)